src/components/sync-dialog.tsx

```diff
@@ -176,6 +177,50 @@ export const useDefaultTree = (): TreeNode[] => {
 	}, [ __ ] );
 };
 
+const WP_LIST_ITEMS: WpListType[] = [ 'plugins', 'themes' ];
+
+const useDynamicTreeState = (
+	type: 'push' | 'pull',
+	localSiteId: string,
+	setTreeState: React.Dispatch< React.SetStateAction< TreeNode[] > >
+) => {
+	const wpListResults = useWpList( localSiteId, WP_LIST_ITEMS );
+
+	useEffect( () => {
+		if ( type === 'pull' ) {
+			return;
+		}
+
+		setTreeState( ( prev ) => {
+			let newState = [ ...prev ];
+
+			const syncOptions = {
+				plugins: SYNC_OPTIONS.plugins,
+				themes: SYNC_OPTIONS.themes,
+			} as const;
+
+			WP_LIST_ITEMS.forEach( ( wpType ) => {
+				const { items, isLoading, error } = wpListResults[ wpType ];
+				const children: TreeNode[] | undefined = error
+					? undefined
+					: items.map( ( item ) => ( {
+							id: item.name,
+							label: item.name,
+							checked: true,
+							type: item.type,
+					  } ) );
+
+				newState = updateNodeById( newState, syncOptions[ wpType ], {
```

Comment:
And use `SYNC_OPTIONS` directly.
```suggestion
				newState = updateNodeById( newState, SYNC_OPTIONS[ wpType ], {
```